crates/store/re_log_types/src/path/parse_path.rs

```diff
@@ -92,29 +89,6 @@ impl std::str::FromStr for DataPath {
                 return Err(PathParseError::TrailingColon);
             }
 
```

Comment:
worth leaving a note why we're not parsing out the archetype information here
... I know @jleibs wants it to be this way here because archetype should be separate optional information, but it actually makes me a bit uneasy. We're still reading out the component type here though, so I'm not sure whether that's correct. Isn't the whole syntax used here now outdated, as we no longer deliminate field names with # ?